### PR TITLE
Refactor local data files browsertests

### PR DESCRIPTION
**Pull request by pilgrim-brave**
## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/extensions/brave_base_local_data_files_browsertest.cc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+void BaseLocalDataFilesBrowserTest::GetTestDataDir(
+    base::FilePath* test_data_dir) {
+  base::ScopedAllowBlockingForTesting allow_blocking;
+  base::PathService::Get(brave::DIR_TEST_DATA, test_data_dir);
+}
+
+void BaseLocalDataFilesBrowserTest::MaybeInitEmbeddedTestServer() {
+  if (embedded_test_server_directory() == NULL)
```
Comment (emerick) [inline browser/extensions/brave_base_local_data_files_browsertest.cc]: Minor nit: `nullptr` here?

File: components/brave_shields/browser/referrer_whitelist_service_browsertest.cc [hunk 2]
```diff
@@ -3,163 +3,98 @@
...
-        g_brave_browser_process->local_data_files_service()->GetTaskRunner()));
-    ASSERT_TRUE(tr_helper->Run());
-    scoped_refptr<base::ThreadTestHelper> io_helper(new base::ThreadTestHelper(
-        base::CreateSingleThreadTaskRunnerWithTraits({BrowserThread::IO})
-            .get()));
-    ASSERT_TRUE(io_helper->Run());
-    base::RunLoop().RunUntilIdle();
+  // BaseLocalDataFilesBrowserTest overrides
+  const char* test_data_directory() override { return kTestDataDirectory; }
+  const char* embedded_test_server_directory() override { return NULL; }
```
Comment (emerick) [inline components/brave_shields/browser/referrer_whitelist_service_browsertest.cc]: Minor nit: `nullptr` here?

File: components/brave_shields/browser/tracking_protection_service_browsertest.cc [hunk 3]
```diff
@@ -34,105 +35,29 @@ const char kRedirectPage[] = "/client-redirect?";
...
-    scoped_refptr<base::ThreadTestHelper> tr_helper(new base::ThreadTestHelper(
-        g_brave_browser_process->local_data_files_service()->GetTaskRunner()));
-    ASSERT_TRUE(tr_helper->Run());
-    scoped_refptr<base::ThreadTestHelper> io_helper(new base::ThreadTestHelper(
-        base::CreateSingleThreadTaskRunnerWithTraits({BrowserThread::IO})
-            .get()));
-    ASSERT_TRUE(io_helper->Run());
+  // BaseLocalDataFilesBrowserTest overrides
+  const char* test_data_directory() override { return kTestDataDirectory; }
+  const char* embedded_test_server_directory() override { return ""; }
```
Comment (emerick) [inline components/brave_shields/browser/tracking_protection_service_browsertest.cc]: Any reason we're returning "" here whereas in the other override we're returning NULL? Seems like we should be consistent if possible.

File: browser/extensions/brave_base_local_data_files_browsertest.cc (hunk 1 above)
Comment (pilgrim-brave) [reply browser/extensions/brave_base_local_data_files_browsertest.cc]: Better to use `!` I think. Will fix.

File: components/brave_shields/browser/tracking_protection_service_browsertest.cc (hunk 3 above)
Comment (pilgrim-brave) [reply components/brave_shields/browser/tracking_protection_service_browsertest.cc]: Better solution I think is to move the files used by the tracking protection browser test to their own subdirectory (they're not used by anything else), then let `embedded_test_server_directory()` return the non-empty path.

**Pull request merged**